modules/apps/27-interchain-accounts/keeper/relay_test.go

```diff
@@ -129,105 +164,361 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 		expPass  bool
 	}{
 		{
-			"Interchain account successfully executes banktypes.MsgSend", func() {
-				// build MsgSend
-				amount, _ := sdk.ParseCoinsNormalized("100stake")
-				interchainAccountAddr, _ := suite.chainB.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), path.EndpointA.ChannelConfig.PortID)
-				msg = &banktypes.MsgSend{FromAddress: interchainAccountAddr, ToAddress: suite.chainB.SenderAccount.GetAddress().String(), Amount: amount}
-				// build packet data
-				data, err := types.SerializeCosmosTx(suite.chainA.Codec, []sdk.Msg{msg})
+			"interchain account successfully executes banktypes.MsgSend",
+			func() {
+				interchainAccountAddr, found := suite.chainB.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), path.EndpointA.ChannelConfig.PortID)
+				suite.Require().True(found)
+
+				msg := &banktypes.MsgSend{
+					FromAddress: interchainAccountAddr,
+					ToAddress:   suite.chainB.SenderAccount.GetAddress().String(),
+					Amount:      sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))),
+				}
+
+				data, err := types.SerializeCosmosTx(suite.chainA.GetSimApp().AppCodec(), []sdk.Msg{msg})
 				suite.Require().NoError(err)
 
 				icaPacketData := types.InterchainAccountPacketData{
 					Type: types.EXECUTE_TX,
 					Data: data,
 				}
+
 				packetData = icaPacketData.GetBytes()
-			}, true,
+			},
+			true,
 		},
 		{
-			"Cannot deserialize packet data messages", func() {
-				data := []byte("invalid packet data")
+			"interchain account successfully executes stakingtypes.MsgDelegate",
+			func() {
+				interchainAccountAddr, found := suite.chainB.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), path.EndpointA.ChannelConfig.PortID)
+				suite.Require().True(found)
+
+				validatorAddr := (sdk.ValAddress)(suite.chainB.Vals.Validators[0].Address)
+				msg := &stakingtypes.MsgDelegate{
+					DelegatorAddress: interchainAccountAddr,
+					ValidatorAddress: validatorAddr.String(),
+					Amount:           sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(5000)),
+				}
+
+				data, err := types.SerializeCosmosTx(suite.chainA.GetSimApp().AppCodec(), []sdk.Msg{msg})
+				suite.Require().NoError(err)
 
 				icaPacketData := types.InterchainAccountPacketData{
 					Type: types.EXECUTE_TX,
 					Data: data,
 				}
+
 				packetData = icaPacketData.GetBytes()
-			}, false,
+			},
+			true,
 		},
 		{
-			"Invalid packet type", func() {
-				// build packet data
-				data, err := types.SerializeCosmosTx(suite.chainA.Codec, []sdk.Msg{&banktypes.MsgSend{}})
+			"interchain account fails to execute stakingtypes.MsgDelegate - insufficient funds",
```

Comment:
Just thinking out loud: do we need to test this? I imagine this is something that the SDK is covering.

Reply:
Yeah it should probably be covered by the SDK I would imagine. I think we can leave it for now and remove later if needs be.